workloads/generator.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+from enum import Enum
+import yaml
+import operator
+import functools
+import random
+import sys
+import json
+
+
+def finished(si, ni):
+    for i in si:
+        if i < ni - 1:
+            return False
+    return True
+
+
+workload_config_file = sys.argv[1]
+output_request_file = sys.argv[2]
+print('loading workload config from: ' + workload_config_file)
+
+with open(workload_config_file) as f:
+    config = f.read()
+
+data = yaml.load(config, Loader=yaml.Loader) # a list of dicts
+
+function_names = [f['name'] for f in data]
+arrival_rates = np.array([f['arrival_rate'] for f in data]) # num of invocations per second
+num_invocations = np.array([f['num_invocation'] for f in data])
+
+print('function names: ' + str(function_names))
+print('arrival rates: ' + str(arrival_rates))
+print('number of invocations: ' + str(num_invocations))
+
+
+mu = 1/arrival_rates
+num_functions = len(arrival_rates)
+num_invocations = num_invocations[0]
+inter_arrival_time = np.random.exponential(mu, (num_invocations, num_functions))
+inter_arrival_time = inter_arrival_time * 1000 # convert to ms
+inter_arrival_time = np.ceil(inter_arrival_time)
+inter_arrival_time_cumsum = np.cumsum(inter_arrival_time, axis=0)
+
+search_index = np.zeros(num_functions, dtype=np.int32)
+
+inter_arrival_time_cumsum = inter_arrival_time_cumsum.astype(int)
+#print(inter_arrival_time_cumsum )
+
+fd = open(output_request_file, 'w')
+
+pmin = 0
+
+while not finished(search_index, num_invocations):
+    candidates = [inter_arrival_time_cumsum[search_index[i], i] for i in range(num_functions)]
+    minv = np.min(candidates)
+    min_idx = np.argmin(candidates)
+
+    interval = minv - pmin
```

Comment:
the interval is the current timestamp minus the last timestamp we just output (or 0 on the first iteration)